### PR TITLE
Handle updated collector instances 

### PR DESCRIPTION
**Pull request by rsvarma95**
- Updated collector.go to handle new collector pods and deleted collector pods (Did not test yet. Will do a test with a sample docker image deployed in the otel operator)
- Removed unnecessary package (suite)

### CODE REVIEW

File: otel-loadbalancer/main.go [hunk 1]
```diff
@@ -130,7 +129,9 @@ func newSharder(ctx context.Context) (*sharder.Sharder, *lbdiscovery.Manager, er
 		sharder.SetTargets(targets)
 		sharder.Reshard()
 	})
-	sharder.SetCollectors(collectors)
+	k8sClient.Watch(ctx, cfg.LabelSelector, func(collectors []string) {
+		sharder.SetCollectors(collectors)
```
Comment (rakyll) [inline otel-loadbalancer/main.go:133]: sharder.SetCollectors is not safe for concurrent usage. We need to protect the collectors with a mutex in the sharder before landing this PR.

Comment (alexperez52) [inline otel-loadbalancer/main.go:133]: Should we also rename this folder* and the places where sharder is used ?

File: otel-loadbalancer/collector/collector.go [hunk 2]
```diff
@@ -2,29 +2,84 @@ package collector
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
```
Comment (rakyll) [inline otel-loadbalancer/collector/collector.go]: Do you know how often things change? watcher.ResultChan() may always return results and we can be stuck in this for loop.

File: otel-loadbalancer/collector/collector.go [hunk 3]
```diff
@@ -2,29 +2,84 @@ package collector
...
+	return Client{
+		k8sClient: clientset,
+	}, nil
+}
+
+func (k Client) Watch(ctx context.Context, labelMap map[string]string, fn func(collectors []string)) {
 	collectors := []string{}
-	pods, err := clientset.CoreV1().Pods(os.Getenv("OTEL_NAMESPACE")).List(context.TODO(), metav1.ListOptions{})
+
+	ns := os.Getenv("OTEL_NAMESPACE")
```
Comment (rakyll) [inline otel-loadbalancer/collector/collector.go]: Maybe, let's put this somewhere package level to make it visible.

**Review by rakyll — CHANGES_REQUESTED** [review]
This is a good start, left some high-level comments.

**Pull request closed without merging**